src/semver.nim

```diff
@@ -0,0 +1,85 @@
+##
+## Copyright (c) 2023, Crash Override, Inc.
+##
+## This file is part of Chalk
+## (see https://crashoverride.com/docs/chalk)
+##
+import strutils, util
+
+# very simple semver implementation
+# it this does NOT handle full semver spec
+# it only handles basic dot-separated version format
+
+type Version* = ref object
+    parts: seq[int]
+
+proc parseVersion*(version: string): Version =
+  new result
+  result.parts = @[]
+  for i in version.strip(chars={'v', ',', '.'}).split('.'):
+    result.parts.add(parseInt(i))
+
+proc `==`*(self: Version, other: Version): bool =
+  for (a, b) in zipLongest(self.parts, other.parts, 0):
+    if a != b:
+      return false
+  return true
+
+proc `!=`*(self: Version, other: Version): bool =
+  return not (self == other)
+
+proc `>`*(self: Version, other: Version): bool =
+  for (a, b) in zipLongest(self.parts, other.parts, 0):
```

Comment:
Again, if you use a tuple, you can just rely on the default tuple comparison, which will be far faster and far more clear.

For instance, something like:

`type Version = tuple[major: uint, minor: uint, patch: uint]`

And then `>` and '!=' both work right without you even having to overload something. And it allows people to clearly ask for `somevar.major`.

Or if you wanted to track any dash and tag name after, you could do it in an object, then just compare the tuple fields.



 (and, more specifically, declare Version

Reply:
changed comparison to use tuple however as in some places it prints original version back keeping it as object so that it can keep the original version name so things like `1.0` dont print as `1.0.0` if the tuple always has 3 versions